src/assets/RestApi/rest-api.md

```diff
@@ -0,0 +1,268 @@
+
+4. **Uniform Interface**: The API has a consistent interface, which simplifies the architecture and enhances visibility of interactions.
+
+## REST API Architecture
+REST APIs utilize a client-server architecture, communicating over HTTP. The client sends requests to the server, which processes them and sends back responses. Communication is stateless, requiring each client request to contain all necessary information for processing.
+
+![REST API](/src/assets/RestApi/images/rest_api.png)
+
+### Components of REST API Architecture
+- **Resources**: In REST, resources are identified by URLs, representing anything that can be named and addressed, such as documents, images, temporal services, collections of other resources, non-virtual objects, and more.
+
+- **HTTP Methods**: REST APIs utilize standard HTTP methods for operations on resources. Common methods include GET (to retrieve a resource), POST (to create a new resource), PUT (to update a resource), and DELETE (to remove a resource).
+
+- **Representations**: Resources are represented in a format easily understood by both client and server, commonly JSON (JavaScript Object Notation) or XML (eXtensible Markup Language).
+
+### Implementing REST API
+Now, lets start to build a REST API. Before we start, make sure you have Python installed on your system. You can download Python from [python.org](https://www.python.org/). We will also need Flask, which you can install using pip:
+```bash
+pip install Flask
+```
+
+We will build a REST API for managing a list of tasks. Each task will have a title and a status indicating whether it's completed or not.
+
+1. Create a new Python file: `app.py`
+
+2. Import Flask and initialize the app:
+```python
+from flask import Flask, jsonify, request
+
+app = Flask(__name__)
+```
+
+3. Define the data structure for our tasks. For simplicity, we'll use a list of dictionaries:
+```python
+tasks = [
+    {"id": 1, "title": "Buy groceries", "completed": False},
+    {"id": 2, "title": "Clean the house", "completed": True}
+]
+```
+
+### GET Requests
+To retrieve all tasks, implement a route that handles GET requests:
+```python
+@app.route("/tasks", methods=["GET"])
+def get_tasks():
+    return jsonify({"tasks": tasks})
+```
+This method will return all the tasks in JSON format.
+
+To retrieve a specific task:
+```python
+@app.route("/tasks/<int:task_id>", methods=["GET"])
+def get_task(task_id):
+    task = next((item for item in tasks if item["id"] == task_id), None)
+    if task is None:
+        return jsonify({"error": "Task not found"}), 404
+    return jsonify({"task": task})
+```
+We use the id of the tasks to identify a specific task since id is unique for every tasks. This method will look for the task with provided id and return the task with the matching id. If the task with the provided id is not found, it will return the error in JSON response indicating the task was not found, with status code `404`.
```

Comment:
if this [comment](https://github.com/JankariTech/blog/pull/123/files#r1567660202) is valid 
then,
include in explanation also